courier/courier.go

```diff
@@ -16,27 +17,38 @@ import (
 
 	gomail "github.com/ory/mail/v3"
 
-	"github.com/ory/kratos/driver/config"
 	"github.com/ory/kratos/x"
 )
 
 type (
-	smtpDependencies interface {
+	SMTPConfig interface {
+		CourierSMTPURL() *url.URL
+		CourierSMTPFrom() string
+		CourierSMTPFromName() string
+		CourierSMTPHeaders() map[string]string
+		CourierTemplatesRoot() string
+	}
+	SMTPDependencies interface {
 		PersistenceProvider
 		x.LoggingProvider
-		config.Provider
+		CourierConfig(ctx context.Context) SMTPConfig
 	}
-	Courier struct {
-		Dialer *gomail.Dialer
-		d      smtpDependencies
+	TemplateTyper            func(t EmailTemplate) (TemplateType, error)
+	EmailTemplateFromMessage func(c SMTPConfig, msg Message) (EmailTemplate, error)
+	Courier                  struct {
+		Dialer                      *gomail.Dialer
+		d                           SMTPDependencies
+		GetTemplateType             TemplateTyper
+		NewEmailTemplateFromMessage EmailTemplateFromMessage
```

Comment:
The idea is that those could be overwritten?

Reply:
Yeah this could be implemented by another application